src/runtime/contrib/clml/clml_runtime.cc

```diff
@@ -146,22 +169,7 @@ class CLMLRuntime : public JSONRuntimeBase {
       LOG(WARNING) << "CLML Runtime Init: Qualcomm extn not present.\n";
       return;
     }
-
-    // Reuse the OpenCl work space from TVM Device API.
-    auto func = tvm::runtime::Registry::Get("device_api.opencl");
-    ICHECK(func != nullptr) << "Cannot find OpenCL device_api in registry";
-    auto device_api = static_cast<cl::OpenCLWorkspace*>(((*func)()).operator void*());
-    this->context = device_api->context;
-    bool queue_found = false;
-    for (size_t i = 0; i < device_api->devices.size(); ++i) {
-      if (device_api->devices[i] == device_id) {
-        this->queue = device_api->queues[i];
-        this->evts = &(device_api->events[i]);
-        queue_found = true;
-      }
-    }
-    ICHECK(queue_found != false) << "Device queue not found in OpenCL Workspace";
-
+    this->queue = GetCommadQueue();
```

Comment:
Sorry, probably I missed something. Could you please explain why you didn't reuse code from OpenCL runtime?
Won't this pseudocode work in the same way?
```suggestion
  void InitCLML() {
    // Setup CLML Context
    cl_int result = 0;
    // Initialize Context and Command Queue
    OpenCLWorkspace* workspace = OpenCLWorkspace::Global();
    workspace->Init();
    OpenCLThreadEntry* t = workspace->GetThreadEntry();
    cl_device_id did = workspace->devices[t->device.device_id];
    
    if (!ExtensionStringPresent(did)) {
      LOG(WARNING) << "CLML Runtime Init: Qualcomm extn not present.\n";
      return;
    }
    this->queue = workspace->GetQueue(t->device);
```

If it works, it won't be necessary to implement function `GetCommadQueue`.

Reply:
Yes, I agree with you. One thing that I wanted to say that probably we could unify and reuse existing code instead of writing new similar functions.
Speaking about recreation function. So, in function `recreateCommandQueue` a new queue is creating and storing in the vector with queues. And when we call  `GetQueue` then a new (recreated) queue will be returned. I don't see any reasons why we cannot call `cl::OpenCLWorkspace::Global()->GetQueue(device)` in all places where we'd like to use command queue and get a new queue.  What do you think about that?

Reply:
Agreed. Let me cleanup a bit by holding ref. for workspace instead of context and queues. This makes CLML unaffected by any changes to workspace internals.